### ISSUE TITLE
will support comfy ?

### ISSUE THREAD

**Issue by kelisiya** [ttchengab/zest_code#2]

**Comment by drphero** [comment]
@kelisiya I've put together a ComfyUI workflow with existing nodes, based on the techniques described in the paper (IPAdapter, depth ControlNet, grayscale subject). I am able to get some decent results, but not for everything. And it requires a positive prompt briefly describing the texture/material and object. I have no clue how the creators were able to do it without prompts. 

Results using sample materials from the paper:

![comparison](https://github.com/ttchengab/zest_code/assets/14174687/1f6ee1ab-d8e8-4d58-9288-90d601b2704b)

Workflow: 
[textureTransfer.json](https://github.com/ttchengab/zest_code/files/14968316/textureTransfer.json)

**Comment by mp3pintyo** [comment]
Thanks for sharing the workflow. The ComfyUI method gave me much better results than the Zero-shot.
![pumpkin-glass-cup](https://github.com/ttchengab/zest_code/assets/5299995/b0911e88-5234-46d0-a2e5-c20e7933687a)
![pumpkin-glass-cup-comfyui](https://github.com/ttchengab/zest_code/assets/5299995/730165ca-ac03-4366-9e1b-d021ca072ef8)
In the case of the zero-shot solution, the edge of the object looks terrible.

**Comment by drphero** [comment]
@kelisiya Well the workflow is basically implementing what they described in the paper and because it's in ComfyUI, it allows for a lot more fine tuning of settings. It's interesting that you liked the result of the glass cup because that is one example I wasn't able to get a good result on in comfy. Your example does have good edges, but the coloring and texture don't really look like the glass cup.

@ttchengab Maybe you could weigh in on how to improve the ComfyUI workflow? For example, how to remove the need for prompting?

**Comment by drphero** [comment]
@kelisiya After playing around a little more, I was able to get this with the standard Realistic Vision V5.1 model

![ComfyUI_temp_uzpzt_00063_](https://github.com/ttchengab/zest_code/assets/14174687/f36b38af-c74e-4158-a2d6-cd2a0677ebd2)

**Comment by mp3pintyo** [comment]
> @kelisiya After playing around a little more, I was able to get this with the standard Realistic Vision V5.1 model
> 

Pretty good! Is the comfyui workflow the same?

**Comment by drphero** [comment]
> Pretty good! Is the comfyui workflow the same?

Yeah, only the prompts were changed I think. The workflow should be embedded in the image.

**Comment by ttchengab** [comment]
> @kelisiya Well the workflow is basically implementing what they described in the paper and because it's in ComfyUI, it allows for a lot more fine tuning of settings. It's interesting that you liked the result of the glass cup because that is one example I wasn't able to get a good result on in comfy. Your example does have good edges, but the coloring and texture don't really look like the glass cup.
> 
> @ttchengab Maybe you could weigh in on how to improve the ComfyUI workflow? For example, how to remove the need for prompting?

Hi @drphero, I am wondering what types of prompts did you add (both positive and negative)? Also, what are the strengths you set for the controlnet/IP-Adaptor? When I was experimenting with different prompts, they tend to have minimal effects (the IP-Adaptor seems to already overwrite this). For the rougher edges of the code I am suspecting it could be due to inaccuracy of DPT in predicting depths, though I am not entirely sure.

Also some suggestions (that may be helpful in trying other projects: the foreground grayscale image brightness can also be tuned for better results. Lift the brightness of the grayscale foreground up for darker images and vice-versa.
